frontend/app/services/SubscriptionService.scala

```diff
@@ -131,12 +133,29 @@ object SubscriptionService {
   }
 }
 
-class SubscriptionService(val tierPlanRateIds: Map[ProductRatePlan, String],
-                          val zuoraSoapService: ZuoraSoapService,
+class SubscriptionService(val zuoraSoapService: ZuoraSoapService,
                           val zuoraRestService: ZuoraRestService) extends AmendSubscription with LazyLogging {
 
   import SubscriptionService._
-  val MembershipProductType = "Membership"
+  val membershipProductType = "Membership"
+  val productRatePlanChargeModel = "FlatFee"
+
+  def tierPlanRateId(productRatePlan: ProductRatePlan): Future[String] =
+    membershipProducts.map { products =>
```

Comment:
Why not extracting the core logic of this method into a pure function? This would make it much easier to test...
```scala
def tierRatePlanId(products: Seq[Rest.Product])(productRatePlan: ProductRatePlan): Option[String]
```